src/rpm/common.go

```diff
@@ -0,0 +1,298 @@
+			if _, err := io.Copy(packed.Data, packValues(value)); err != nil {
+				return nil, err
+			}
+		}
+
+	case rpmTypeInt32: // INT32
+		// value should be []int32
+		int32Values, ok := tag.Value.([]int32)
+		if !ok {
+			return nil, fmt.Errorf("INT32 value should be []int32")
+		}
+
+		packed.Count = len(int32Values)
+
+		for _, value := range int32Values {
+			if _, err := io.Copy(packed.Data, packValues(value)); err != nil {
+				return nil, err
+			}
+		}
+
+	case rpmTypeInt64: // INT64
+		// value should be []int64
+		int64Values, ok := tag.Value.([]int64)
+		if !ok {
+			return nil, fmt.Errorf("INT64 value should be []int64")
+		}
+
+		packed.Count = len(int64Values)
+
+		for _, value := range int64Values {
+			if _, err := io.Copy(packed.Data, packValues(value)); err != nil {
+				return nil, err
+			}
+		}
+
+	default:
+		return nil, fmt.Errorf("Unknown tag type: %d", tag.Type)
+	}
+
+	return &packed, nil
+}
+
+// header_index_record
+// https://github.com/tarantool/cartridge-cli/blob/cafd75a5c8ddfdb93ef8290d6e4ebd5d83e4c46e/cartridge-cli.lua#L2041
+func getPackedTagIndex(offset int, tagID int, tagType rpmValueType, count int) *bytes.Buffer {
+	tagIndex := packValues(
+		int32(tagID),
+		int32(tagType),
+		int32(offset),
+		int32(count),
+	)
+
+	return tagIndex
+}
+
+// gen_index_header
+// https://github.com/tarantool/cartridge-cli/blob/cafd75a5c8ddfdb93ef8290d6e4ebd5d83e4c46e/cartridge-cli.lua#L2033
+func getTagSetHeader(tagsNum int, dataLen int) *bytes.Buffer {
+	tagSetHeader := packValues(
+		headerMagic,
+		byte(versionMagic),
+		int32(reservedMagic),
+		int32(tagsNum),
+		int32(dataLen),
+	)
+
+	return tagSetHeader
+}
+
+// gen_header
+// https://github.com/tarantool/cartridge-cli/blob/cafd75a5c8ddfdb93ef8290d6e4ebd5d83e4c46e/cartridge-cli.lua#L1961
+func packTagSet(tagSet rpmTagSetType, regionTagID int) (*bytes.Buffer, error) {
+	var resData = bytes.NewBuffer(nil)
+	var tagsIndex = bytes.NewBuffer(nil)
+	var resIndex = bytes.NewBuffer(nil)
+
+	// tags index
+	for _, tag := range tagSet {
+		packed, err := packTag(tag)
+
+		if err != nil {
+			return nil, err
+		}
+		if boundaries, ok := boundariesByType[tag.Type]; !ok {
+			return nil, fmt.Errorf("Boundaries for type %d is not set", tag.Type)
+		} else if boundaries > 1 {
+			alignData(resData, boundaries)
+		}
+
+		tagIndex := getPackedTagIndex(resData.Len(), tag.ID, tag.Type, packed.Count)
+
+		if err := common.ConcatBuffers(resData, packed.Data); err != nil {
+			return nil, err
+		}
+
+		if err := common.ConcatBuffers(tagsIndex, tagIndex); err != nil {
+			return nil, err
+		}
+	}
+
+	// regionTag index
+	regionTagIndex := getPackedTagIndex(resData.Len(), regionTagID, rpmTypeBin, 16)
+
+	// resIndex is regionTagIndex + tagsIndex
+	if err := common.ConcatBuffers(resIndex, regionTagIndex, tagsIndex); err != nil {
+		return nil, err
+	}
+
+	// regionTag data
+	tagsNum := len(tagSet) + 1
+	regionTagData := getPackedTagIndex(-tagsNum*16, regionTagID, rpmTypeBin, 16)
```

Comment:
nit: `-tagsNum*16` -> `-tagsNum * 16`... codestyle...

Reply:
Go formatter thinks it's OK.